WordPress/gutenberg
# Fix: Vertical toolbar placement in Zoom Out mode

## What?

This PR fixes the issue [#65188](https://github.com/WordPress/gutenberg/issues/65188)
It updates the Popover component (used by the zoom out toolbar) to observe changes in the `.interface-interface-skeleton__body` element and trigger position updates accordingly.
The `.interface-interface-skeleton__body` element is the parent div of the inserter.


## Why?

The current implementation does not update the popover toolbar's position when the inserter menu is opened in the zoom out mode. This PR ensures that the popover toolbar's position is updated correctly when layout changes occur to the parent inserter.

## How?

- Added a `MutationObserver` to observe changes in the parent div of the inserter.
- Updated the `whileElementsMounted` function (which updates the toolbar's position relative to the reference element) to include the parent div of the inserter as a reference for useFloating.

## Testing Instructions

1. Enable experimental zoom out mode from Gutenberg settings. 
2. Open a post or page.
3. Insert a block or select an existing block.
4. Enter zoom out mode.
5. Verify if the floating toolbar is present, if not select a block again.
6. Now open the inserter menu (+ icon in the top left).
7. Verify that the floating toolbar is not getting detached from the block.


### Testing Instructions for Keyboard

1. Enable experimental zoom out mode from Gutenberg settings. 
2. Open a post or page.
3. Insert a block or select an existing block.
4. Enter zoom out mode using tab to go to the zoom but button (top right).
5. Verify if the floating toolbar is present, if not select a block again.
6. Now open the inserter menu using tab key to navigate keyboard (+ icon in the top left).
7. Verify that the floating toolbar is not getting detached from the block.


## Screenshots or screencast 
Before:
<img width="380" alt="image" src="https://github.com/user-attachments/assets/c507870e-24e4-4cee-bcba-9caff9ee54ee">


After:
<img width="380" alt="image" src="https://github.com/user-attachments/assets/24d392da-c690-4da7-95ca-56f0cd9e9ca2">
